CRM/Contact/BAO/Query.php

```diff
@@ -5164,15 +5164,20 @@ public function dateQueryBuilder(
         $date = "'$date'";
       }
 
+      $this->_tables[$tableName] = $this->_whereTables[$tableName] = 1;
+
+      // handle IS NULL / IS NOT NULL / IS EMPTY / IS NOT EMPTY
+      if ($this->nameNullOrEmptyOp($fieldName, $op, $grouping)) {
```

Comment:
Yashodha, I've moved the code earlier, but not as early as suggested. There are two reasons it can't be moved earlier

1. The value of $op may be changed at line 5141
2. The value of $name might take on a value other than $fieldName or "{$fieldName}_low" or "{$fieldName}_high". See line 3802 for an example.